### ISSUE TITLE
Cannot create Simulators on M1 helix instances

### ISSUE THREAD

**Issue by steveisok** [dotnet/xharness#535]
In trying an arm64 simulator test run on CI, I noticed Xharness is not able to find any simulator to launch.  My guess is that they just need to be installed.  

```
[10:29:04] dbug: 10:29:04.6999790 Unable to create a device for device type: iPhone X (com.apple.CoreSimulator.SimDeviceType.iPhone-X), runtime: iOS 14.1 (14.1 - 18A8394) - com.apple.CoreSimulator.SimRuntime.iOS-14-1 (unavailable, failed to open liblaunch_sim.dylib)
[10:29:04] dbug: 10:29:04.7011280 Process simctl exited with 145
[10:29:04] dbug: 10:29:04.7032010 Could not create device

[10:29:04] dbug: 10:29:04.8313850 An error was encountered processing the command (domain=com.apple.CoreSimulator.SimError, code=401):
[10:29:04] dbug: 10:29:04.8316260 The iOS 14.1 simulator runtime is not available.
[10:29:04] dbug: 10:29:04.8316480 failed to open liblaunch_sim.dylib
[10:29:04] dbug: 10:29:04.8316580 Unable to open liblaunch_sim.dylib.  Try reinstalling Xcode or the simulator runtime.
```

Here's the helix log for your reference: 

https://helixre107v0xdeko0k025g8.blob.core.windows.net/runtime-dotnet-a4f46f68016a43239c/System.Buffers.Tests/console.a860aa26.log?sv=2019-07-07&se=2021-04-22T17%3A28%3A30Z&sr=c&sp=rl&sig=S2Qkd%2BfCt2nqoUBHaE%2B2hxDkd9dtiRAu5wB0siq%2FHPk%3D

**Comment by premun** [comment]
@steveisok I found out that we need Xcode 12.4 while we only have 12.1. I successfully ran an XHarness job using 12.4.

I opened https://github.com/dotnet/core-eng/issues/12757

**Comment by premun** [comment]
Will be resolved by dotnet/core-eng#12757

**Comment by premun** [comment]
@steveisok the queue has been updated and the Simulators should work now.
